src/App.js

```diff
@@ -29,37 +30,113 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player1, togglePlayer1] = useState(true);
+  const [gameover, checkGame] = useState(false);
+
+  const togglePlayers = () => {
+    if (player1 === true) {
+      togglePlayer1(false);
+    } else {
+      togglePlayer1(true);
+    }
+  };
+  const onClickCallback = (squareToMark) => {
+    // console.log(squareToMark);
+    const newboard = [...squares];
+    // console.log(newboard);
+    if (!gameover) {
+      newboard.map((row) => {
+        row.map((square) => {
+          if (square.id === squareToMark) {
+            // console.log(squareToMark);
+            if (player1 === true) {
+              square.value = 'x';
+              togglePlayers();
+              return {
+                id: square.id,
+                value: 'x',
+                key: square.id,
+              };
+            } else {
+              square.value = 'o';
+              togglePlayers();
+              console.log(square);
+              return square;
+            }
+          }
+          return square;
+        });
+      });
+    }
```

Comment:
In the inner `map` loop, you are returning the square twice, once within the `if/else` and again at the end of each iteration. We can refactor by omitting the `return` inside of the `if/else`.

```suggestion
    if (!gameover) {
      newboard.map((row) => {
        row.map((square) => {
          if (square.id === squareToMark) {
            // console.log(squareToMark);
            if (player1 === true) {
              square.value = 'x';
              togglePlayers();
            } else {
              square.value = 'o';
              togglePlayers();
            }
          }
          return square;
        });
      });
    }
```